README.md

```diff
@@ -348,20 +369,20 @@ type IBody = {
 	age: number
 	meta: {
 		canFly: boolean
-		pickupItems: ("egg" | "gras" | "stone")[]
+		pickupItems: ("egg" | "grass" | "stone")[]
 	}
 }
 ```
 
-In some cases, for complex or deeply nested checkers, typescript is not able to infer the type (anymore). In this cases, or if you prefer to declare your interfaces and types separately, you can just provide the type to the checker to make sure your types and checkers are in sync.
+If you prefer to declare your interfaces and types separately, you can just provide the type to the checker to make sure your checked types ar assignable to the interface types ahead of time. This can't ensure equivalence for optional members and union types.
```

Comment:
```suggestion
If you prefer to declare your interfaces and types separately, you can just provide the type to the checker to make sure your checked types are assignable to the interface types ahead of time. This can't ensure equivalence for optional members and union types.
```